observable-slim.js

```diff
@@ -191,7 +191,20 @@ var ObservableSlim = (function() {
 				
 				// record the deletion that just took place
 				changes.push({"type":"delete","target":target,"property":property,"newValue":null,"previousValue":previousValue[property],"currentPath":currentPath,"proxy":proxy});
-				
+
+                var t = targets.indexOf(target);
+                if (t > -1) {
+                    var j = targetsProxy[t].length;
+                    while (j--) {
+                        var beforeChange = targetsProxy[t][j].observable.beforeChange;
+                        if(typeof beforeChange === 'function') {
+                            var res = beforeChange(changes);
+                            if (res === false)
+                                return false;
+                        }
+                    }
+                }
```

Comment:
There's some mixing of tabs and spaces. Your new lines are using spaces for indenting, but the existing code uses tabs. The code might look like its lined up in your editor, but if you take a look on Github you'll see that the indenting is misaligned due to mixing of tabs and spaces.